### ISSUE TITLE
Provide access to other field values

### ISSUE THREAD

**Issue by thomasbaustert** [dryrb/dry-validation#63]
Thanks for dry-validation! Can you provide access to others field values (e.g. value(key)) for something like this? Or is it already possible?

    key(:start_date)    { |date| date.filled? & date.date? }
    key(:end_date)      { |date| date.filled? & date.date? }
    key(:date_of_issue) { |date| date.filled? & date.date? & date.in_period? }

    def in_period?(value)
      # start_date = coerced_value(:start_date)
      # end_date   = coerced_value(:end_date)
      # date       = coerced_value(:date_of_issue)
      # value(key) returns raw value
      start_date = Date.parse(value(:start_date))
      end_date   = Date.parse(value(:end_date))      
      date       = Date.parse(value)
      (start_date..end_date).include?(value)
    end

**Comment by solnic** [comment]
@thomasbaustert yes, this will be supported in the next version

**Comment by thomasbaustert** [comment]
Thanks!

**Comment by solnic** [comment]
to be more specific, this will be possible:

``` ruby
key(:start_date).required(:date?)
key(:end_date).required(:date?)
key(:date_of_issue).required(:date?)

rule(date_of_issue: :in_period?) do
  value(:date_of_issue).between?(value(:start_date), value(:date_of_issue))
end
```

**Comment by solnic** [comment]
This is now possible in master and will be available in 0.7.0.
